cg/apps/janus/api.py

```diff
@@ -0,0 +1,28 @@
+"""The JanusAPIClient."""
+
+from http import HTTPStatus
+
+import requests
+from requests import Response
+
+from cg.apps.janus.dto.create_collect_qc_request import CreateCollectQCRequest
+from cg.exc import JanusAPIError
+
+
+class JanusAPIClient:
+    def __init__(self, config: dict[str]):
+        self.host = config["janus"]["host"]
+
+    def get_qc_metrics(self, collect_qc_request: CreateCollectQCRequest):
+        endpoint: str = f"{self.host}/collect_qc_metrics"
+        response = requests.post(endpoint, data=collect_qc_request)
+        if response.status_code == HTTPStatus.OK:
+            return response.json()
+        self._handle_errors(response)
+
+    @staticmethod
+    def _handle_errors(response: Response):
+        if 400 <= response.status_code < 500:
+            raise JanusAPIError(f"Client error: {response.status_code}")
+        elif 500 <= response.status_code:
+            raise JanusAPIError(f"Server error: {response.status_code}")
```

Comment:
I've added this to the pattern, basically it is better to abstract away the `HTTP` specific errors so that the user of the client does not need to know about them. A single base error for the client should be defined and then specific exceptions inheriting from that exception can be raised. Like `JanusClientInvalidAuthentication` or something

Reply:
Hmm, I don't really see the value in this. Could you give an illuminating example of when it would make a user unhappy? Like in both of these cases you could just catch the parent class HTTPError?

Reply:
Yes, it is true that we could just catch the HTTP error instead. But I think it is good design to map the errors like this because
- The user of the client does not need to know that the client uses HTTP
- If an exception bubbles up, it is better that its origin is clear rather than being just an HTTPexception

Reply:
Will use the `_handle_errors` for now as it adheres closest to the cg-ep. We can revert back if discussions alter the proposal.